src/server/scripts/EasternKingdoms/MagistersTerrace/boss_felblood_kaelthas.cpp

```diff
@@ -37,31 +29,88 @@ enum Says
 enum Spells
 {
     // Phase 1
-    SPELL_FIREBALL                  = 44189,
-    SPELL_FLAMESTRIKE_SUMMON        = 44192,
-    SPELL_PHOENIX                   = 44194,
-    SPELL_SHOCK_BARRIER             = 46165,
-    SPELL_PYROBLAST                 = 36819,
+    SPELL_FIREBALL                 = 44189,
+    SPELL_FLAMESTRIKE_SUMMON       = 44192,
+    SPELL_PHOENIX                  = 44194,
+    SPELL_SHOCK_BARRIER            = 46165,
+    SPELL_PYROBLAST                = 36819,
 
     // Phase 2
-    SPELL_SUMMON_ARCANE_SPHERE      = 44265,
-    SPELL_TELEPORT_CENTER           = 44218,
-    SPELL_GRAVITY_LAPSE_INITIAL     = 44224,
-    SPELL_GRAVITY_LAPSE_PLAYER      = 44219, // Till 44223, 5 players
-    SPELL_GRAVITY_LAPSE_FLY         = 44227,
-    SPELL_GRAVITY_LAPSE_DOT         = 44226,
-    SPELL_GRAVITY_LAPSE_CHANNEL     = 44251,
-    SPELL_POWER_FEEDBACK            = 44233
+    SPELL_SUMMON_ARCANE_SPHERE     = 44265,
+    SPELL_TELEPORT_CENTER          = 44218,
+    SPELL_GRAVITY_LAPSE_INITIAL    = 44224,
+    SPELL_GRAVITY_LAPSE_PLAYER     = 44219, // Till 44223, 5 players
+    SPELL_GRAVITY_LAPSE_FLY        = 44227,
+    SPELL_GRAVITY_LAPSE_DOT        = 44226,
+    SPELL_GRAVITY_LAPSE_CHANNEL    = 44251,
+    SPELL_POWER_FEEDBACK           = 44233,
+
+    // Phoenix spells
+    SPELL_PHOENIX_BURN             = 44197,
+    SPELL_REBIRTH                  = 44196
 };
 
 enum Misc
 {
-    ACTION_TELEPORT_PLAYERS     = 1,
-    ACTION_KNOCKUP              = 2,
-    ACTION_ALLOW_FLY            = 3,
-    ACTION_REMOVE_FLY           = 4,
+    ACTION_TELEPORT_PLAYERS    = 1,
+    ACTION_KNOCKUP             = 2,
+    ACTION_ALLOW_FLY           = 3,
+    ACTION_REMOVE_FLY          = 4,
 
-    CREATURE_ARCANE_SPHERE      = 24708
+    CREATURE_ARCANE_SPHERE     = 24708,
+    CREATURE_PHOENIX           = 24674
+};
+
+struct npc_phoenix_tk : public ScriptedAI
+{
+    npc_phoenix_tk(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        _scheduler.CancelAll();
+        me->SetReactState(REACT_AGGRESSIVE);
+
+        _scheduler.Schedule(5s, [this](TaskContext context)
+        {
+            DoCastSelf(SPELL_PHOENIX_BURN);
+            context.Repeat(10s);
+        });
+
+        _scheduler.Schedule(2s, [this](TaskContext context)
+        {
+            if (Creature* kael = ObjectAccessor::GetCreature(*me, _kaelGuid))
+            {
+                if (kael->GetHealthPct() <= 50.0f)
+                {
+                    DoCastRandomTarget(SPELL_FIREBALL, 0, 40.0f);
+                }
+            }
```

Comment:
dont use brackets one oneliners. goes for the whole file.

use HealthBelowPct instead of GetHealthPct